pinot-core/src/main/java/com/linkedin/pinot/core/plan/TransformPlanNode.java

```diff
@@ -0,0 +1,171 @@
+    List<String> columns = new ArrayList<>();
+    for (TransformExpressionTree expressionTree : expressionTrees) {
+      expressionTree.getColumns(columns);
+    }
+    return columns;
+  }
+
+  /**
+   * Helper method to build TransformExpressionTrees for a given set of transform expressions.
+   *
+   * @param transformExpressions List of expressions for which to build trees.
+   * @param ignoreColumns If true, expressions that are just column names are ignored.
+   * @return List of expression trees for the given transform expressions.
+   */
+  public static List<TransformExpressionTree> buildTransformExpressionTrees(List<String> transformExpressions,
+      boolean ignoreColumns) {
+    List<TransformExpressionTree> expressionTrees = new ArrayList<>(transformExpressions.size());
+
+    for (String transformExpression : transformExpressions) {
+      TransformExpressionTree expressionTree = _compiler.get().compileToExpressionTree(transformExpression);
+      if (!ignoreColumns || !expressionTree.isColumn()) {
+        expressionTrees.add(expressionTree);
+      }
+    }
+    return expressionTrees;
+  }
+
+  /**
+   * Helper method to extract projection columns and transform expressions from the given
+   * BrokerRequest.
+   *
+   * @param brokerRequest BrokerRequest to process
+   * @param projectionColumns Output projection columns from broker request
+   * @param transformExpressions Output transform expression from broker request
+   */
+  private void extractColumnsAndTransforms(BrokerRequest brokerRequest, List<String> projectionColumns,
+      List<String> transformExpressions) {
+
+    if (brokerRequest.isSetAggregationsInfo()) {
+      for (AggregationInfo aggregationInfo : brokerRequest.getAggregationsInfo()) {
+        if (!aggregationInfo.getAggregationType()
+            .equalsIgnoreCase(AggregationFunctionFactory.AggregationFunctionType.COUNT.getName())) {
+          String columns = aggregationInfo.getAggregationParams().get("column").trim();
+          projectionColumns.addAll(Arrays.asList(columns.split(",")));
+        }
+      }
+
+      // Collect all group by related columns.
+      if (brokerRequest.isSetGroupBy()) {
+        GroupBy groupBy = brokerRequest.getGroupBy();
+        List<String> groupByColumns = groupBy.getColumns();
+
+        // Only transforms in group-by
+        if (groupByColumns != null) {
+          projectionColumns.addAll(groupByColumns);
+        }
+
+        // Check null for backward compatibility.
+        List<String> expressions = groupBy.getExpressions();
+        if (expressions != null) {
+          transformExpressions.addAll(expressions);
+          if (groupByColumns != null && !groupByColumns.isEmpty()) {
+            transformExpressions.removeAll(groupByColumns);
+          }
+        }
+      }
+    } else {
+      // TODO: Support transforms for selection.
+      projectionColumns.addAll(brokerRequest.getSelections().getSelectionColumns());
```

Comment:
No, this will be a pass through to non-transform code in absence of transforms. In presence of transforms, the code will behave the same way as it does today and throw exception wherever it does that today.